File: Java-Spring/src/main/java/org/codecop/lcdnumbers/ScalingRepeaterImpl.java
package org.codecop.lcdnumbers;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import org.springframework.stereotype.Service;

/**
 * Repeat objects and characters according to scaling.
 */
@Service
public class ScalingRepeaterImpl implements ScalingRepeater {

    @Override
    public <T> List<T> repeat(T element, Scaling scaling) {
        List<T> elements = new ArrayList<>();
        scaling.times(() -> elements.add(element));
        return elements;
    }

    @Override
    public String repeat(char aChar, Scaling scaling) {
        StringBuilder acc = new StringBuilder();
        scaling.times(() -> acc.append(aChar));
        return acc.toString();
    }
}

File: Kotlin-Spring/src/main/java/org/codecop/lcdnumbers/ScalingRepeaterImpl.kt
package org.codecop.lcdnumbers

import org.springframework.stereotype.Service
import java.util.ArrayList
import java.util.Objects

/**
 * Repeat objects and characters according to scaling.
 */
@Service
class ScalingRepeaterImpl : ScalingRepeater {
    override fun <T> repeat(element: T, scaling: Scaling): List<T> {
        Objects.requireNonNull(element)
        Objects.requireNonNull(scaling)
        val elements: MutableList<T> = ArrayList()
        scaling.times(Runnable { elements.add(element) })
        return elements
    }
    
    override fun repeatChar(aChar: Char, scaling: Scaling): String {
        Objects.requireNonNull(scaling)
        val acc = StringBuilder()
        scaling.times(Runnable { acc.append(aChar) })
        return acc.toString()
    }
}